components/apimgt-extensions/org.wso2.carbon.apimgt.webapp.publisher/src/main/java/org/wso2/carbon/apimgt/webapp/publisher/APIPublisherService.java

```diff
@@ -39,7 +40,7 @@
      * @param api An instance of the bean that passes metadata related to the API being published
      * @throws APIManagementException Is thrown if some unexpected event occurs while publishing the API
      */
-    void publishAPI(API api) throws APIManagementException;
+    void publishAPI(API api) throws APIManagementException, FaultGatewaysException;
```

Comment:
Shouldn't FaultGatewaysException be hidden from APIPublisherService interface? What we should probably do is wrapping the same with an instance of APIManagementException so that what gets exposed outside via the interface is only "APIManagementException".